internal/configsource/helpers.go

```diff
@@ -0,0 +1,43 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configsource
+
+// WatcherNotSupported is the a watcher function that always returns ErrWatcherNotSupported.
+func WatcherNotSupported() error {
+	return ErrWatcherNotSupported
```

Comment:
Is this supported to be used by sources? Why not just return ErrWatcherNotSupported? Doesn't look like this will help much.

Reply:
Yeah, that is very trivial to implement but I found myself repeating that code many times. I was repeating that in a few places - at least vault and an environment config source will use it. Let's re-evaluate as we bring these to the repo.